.github/workflows/workflows.yaml

```diff
@@ -89,11 +90,15 @@ jobs:
         id: queued
         with:
           sha: ${{ github.event.workflow_run.head_sha || github.event.pull_request.head.sha || inputs.ref || github.sha }}
-          name: workflows / ${{ github.job }} (${{ matrix.path }})
+          name: workflows / ${{ github.job }} (${{ matrix.path }}) (create-check on ${{ (github.event.workflow_run && 'workflow_run') || (github.event.pull_request && 'pull_request') || (inputs && 'workflow_dispatch') }})
           job-name: ${{ github.job }} (${{ matrix.path }})
-          if: ${{ !!github.event.workflow_run }}
+          if: ${{ !!github.event.workflow_run || (github.event.pull_request && !github.event.pull_request.head.repo.fork) }}
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          ref: ${{ github.event.workflow_run.head_sha || github.event.pull_request.head.sha || inputs.ref || github.sha }}
       - name: Setup Custard
-        uses: GoogleCloudPlatform/cloud-samples-tools/actions/steps/setup-custard@v0.2.4
+        uses: GoogleCloudPlatform/cloud-samples-tools/actions/steps/setup-custard@57945e1e95917407558192c975ccc1e1f1e7a0e7
```

Comment:
Changing this to a hash may break release-please/renovatebot updates, to keep in mind next release if this isn't picked up

Reply:
It's the only workaround I've found to test the changes in the PR. We should come up with a better strategy. We can change it back to v0.2.4 before merging, it'll just cause test failures since it'll be fetching the wrong actions

Reply:
That sha links to a previous commit in this PR, I didn't realise that. I would suggest when making these temporary workarounds you add a comment to ensure communication about reversion before merging

Reply:
This is, in general, something that we should figure out. Even with this workaround, once we merge it, tests will be broken in main until a new tag is released.